### PR TITLE
Change SSH version field to be `&optional`.

### LINKED ISSUE #1590: Changes to the behavior of the SSH version field were not documented
https://github.com/zeek/zeek/commit/3769ed6c6687e5ecf93c3668bebe3c3f7da3d0fb started introducing 0 to the ssh version field for client\server mismatches without documenting the change in behavior.

### PR DESCRIPTION
**Pull request by bbannier**
In 3769ed6c668 we added handling for SSH version 1.99 which unsed a SSH
version of 0 to indicate weird cases where no version could be
determined.

This patch is a fixup for that patch. Instead of using a magic version
of 0 we now use an `&optional` version value. If no SSH version can be
extracted the version will be unset; additionally a `conn_weird` event
will be raised.

Closes #1590.

### CODE REVIEW

**Review by grigorescu — COMMENTED** [review]
Looks good to me!

**Review by 0xxon — CHANGES_REQUESTED** [review]

File: scripts/base/protocols/ssh/main.zeek [hunk 1]
```diff
@@ -20,8 +20,9 @@ export {
 		uid:             string       &log;
 		## The connection's 4-tuple of endpoint addresses/ports.
 		id:              conn_id      &log;
-		## SSH major version (1 or 2)
-		version:         count        &log;
+		## SSH major version (1, 2, or unset). If no version can be set a
+		## `conn_weird` event will be raised.
```
[the remark this answers is not in the document]
Comment (bbannier) [reply scripts/base/protocols/ssh/main.zeek]: I expanded this part (also incorporating the changes you asked for below).

File: scripts/base/protocols/ssh/main.zeek [hunk 2]
```diff
@@ -156,51 +157,58 @@ function set_session(c: connection)
...
+		# SSH2 vs SSH2
+		else if (c$ssh$client[4] == "2" && c$ssh$server[4] == "2" )
+			{
+			c$ssh$version = 2;
+			}
+
+		return;
+		}
+
+	Reporter::conn_weird("SSH protocol version mismatch", c, version);
```
[the remark this answers is not in the document]
Comment (bbannier) [reply scripts/base/protocols/ssh/main.zeek]: I tried to address this. Now we only raise a `conn_weird` if both client and server versions are set to some non-empty strings, and no shared version can be extracted. I also tweaked the message passed in the weird (let me know if the way I pass up the client and server version strings isn't useful), and used different messages for the different paths.

**Comment by 0xxon** [comment]
Just FYI - I renamed the weirds during the merge.

**Pull request merged**